### PR TITLE
BUG: setting td64 value into numeric Series incorrectly casting to int

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

We fixed this a few days ago for some of the setting-methods, but there were a few we missed.  Re-using the TestSetitemEquivalents pattern is much more thorough.

We'll be able to de-duplicate the test class following #39477

### CODE REVIEW

**Comment by jreback** [comment]
> We'll be able to de-duplicate the test class following #39477

this is deduplicated?

**Comment by jbrockmendel** [comment]
> this is deduplicated?

haven't updated to de-dup yet, turns out to be a little non-trivial bc this makes some view-or-not assertions the other doesnt

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4331,7 +4331,15 @@ def where(self, cond, other=None):
         except (ValueError, TypeError):
             return self.astype(object).where(cond, other)
 
-        values = np.where(cond, values, other)
+        if isinstance(other, np.timedelta64) and self.dtype == object:
+            # https://github.com/numpy/numpy/issues/12550
+            #  timedelta64 will incorrectly cast to int
```
Comment (jreback) [inline pandas/core/indexes/base.py:4336]: too bad you cannot use np.where then cast back the integers if necessary

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
prob should add a whatsnew for this change

**Pull request merged**